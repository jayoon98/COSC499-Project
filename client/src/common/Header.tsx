import React from 'react';
import { Title } from './Title';
import { View, StyleSheet, ScrollView, Dimensions } from 'react-native';

const screen = Dimensions.get('screen');

export type HeaderProps = {
  title: string;
  children?: React.ReactNode;
};

export function Header({ title, children }: HeaderProps) {
  return (
    <View style={styles.header}>
      <Title style={{ fontSize: screen.height * 0.035 }}>{title}</Title>
      <View style={{ flex: 1, flexDirection: 'row' }}>{children}</View>
    </View>
  );
}

const styles = StyleSheet.create({
  header: {
    height: screen.height * 0.12,
    justifyContent: 'flex-start',
    display: 'flex',

    paddingLeft: 12,
    paddingRight: 12,
    paddingBottom: 2,
    borderBottomWidth: 1,
    borderBottomColor: '#f2f6fa',
    flexDirection: 'row',
    alignItems: 'flex-end',
  },
});
