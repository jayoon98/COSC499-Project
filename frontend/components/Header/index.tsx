import Link from 'next/link';
import { useRouter } from 'next/router';
import useLocalStorage from '../../functions/useLocalStorage';

export default function Header() {
  const [auth, setAuth] = useLocalStorage('auth', { email: null, authkey: null });
  const [prefs] = useLocalStorage('prefs', {
    first_name: null,
    last_name: null,
    date_created: null,
    last_login: null,
    isClient: null,
    isAdmin: null,
    circle_colors: null,
    circle_rank: null,
    notify_time: null,
    avatar: null,
  });
  const router = useRouter();

  function logout() {
    setAuth({
      email: null,
      authkey: null,
    });
    router.push('/');
  }

  return (
    <>
      <div className="header">
        {auth.email != null ? (
          <Link href="/profile">
            <p className="header-email">{auth.email}</p>
          </Link>
        ) : <></>}
        <div className="header-container">
          <Link href="/">
            <div className="header-button">Home</div>
          </Link>
          {prefs.isAdmin ? (
            <Link href="/adminPanel">
              <p className="header-button"> Admin Panel</p>
            </Link>
          ) : <></>}
          {auth.email != null ? (
            <>
              <Link href="/profile">
                <div className="header-button">Profile</div>
              </Link>
              <Link href="/domains">
                <div className="header-button">My Domains</div>
              </Link>
            </>
          ) : <></>}
          <Link href="/crisis">
            <div className="header-button">Are You In Crisis?</div>
          </Link>
          <Link href="/information">
            <div className="header-button">Information</div>
          </Link>
          <Link href="/contact">
            <div className="header-button">Contact Therapist</div>
          </Link>
          {auth.email == null ? (
            <>
              <Link href="/login">
                <div className="header-button">Login</div>
              </Link>
              <Link href="/signup">
                <div className="header-button">Sign Up</div>
              </Link>
            </>
          ) : (
            <>
              <div role="button" tabIndex={0} className="header-button" onClick={() => logout()} onKeyDown={() => logout()}>
                Logout
              </div>
            </>
          )}
        </div>
      </div>
    </>
  );
}
