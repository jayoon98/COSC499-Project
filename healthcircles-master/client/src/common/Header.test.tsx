import React from 'react';
import { Header } from './Header';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<Header title="Hi" />).toJSON();
  expect(tree).toMatchSnapshot();
});
