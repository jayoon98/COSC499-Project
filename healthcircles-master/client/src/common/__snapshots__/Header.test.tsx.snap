// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "borderBottomColor": "#f2f6fa",
      "borderBottomWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "height": 160.07999999999998,
      "justifyContent": "flex-start",
      "paddingBottom": 2,
      "paddingLeft": 12,
      "paddingRight": 12,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 60.03,
        "fontWeight": "bold",
      }
    }
  >
    Hi
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  />
</View>
`;
